### ISSUE TITLE
 Recieved invalid apReq, exploit will fail

### ISSUE THREAD

**Issue by Huk4aro** [Dec0ne/KrbRelayUp#2]
Tested against a few Windows10 machines, Couldn't seem to solve this.
LDAP Server Signing is not enabled.
I'm not sure what's causing this, Looking for a fix right now
![Screenshot 2022-04-27 013723](https://user-images.githubusercontent.com/96992584/165404413-8be76dd7-1d03-42fc-a3d3-380d065f63b3.png)

**Comment by bestrocker221** [comment]
Same exact Issue here. LDAP signing disabled

**Comment by Neo23x0** [comment]
Same here

**Comment by 0xBrAinsTorM** [comment]
Got it working in an old lab setup (https://github.com/clong/DetectionLab) which still was idling around. It's not working on another lab i have. 

![grafik](https://user-images.githubusercontent.com/26419565/165526174-1ada92da-1602-49d2-9e80-0e3b488d71cf.png)

Unfortunately i don't have the time to investigate the differences.

**Comment by nodauf** [comment]
There could be multiple issues. But in my case, I compared two pcaps and in the one where the error `Received invalid apReq, exploit will fail` was returned, the coerced authentication was NTLM authentication instead of kerberos.

The difference is clear in the packet `Bind: call_id: 2, Fragment: Single, 3 context items: IRemUnknown2 V0.0 ...`

When this error is returned, the packet looks like this:
![image](https://user-images.githubusercontent.com/7670696/165893578-2a5ef4ac-e379-46ef-86d3-cef2c6362067.png)

When the tool is working as intended, the packet looks like this:
![image](https://user-images.githubusercontent.com/7670696/165893752-98463b83-2152-47e7-a556-47a96c7a97ec.png)

I have no idea why but it may help someone else to dig in the right direction

**Comment by kinomakino** [comment]
Same scenario whit this tool and krbrelay (main)

**Comment by Dec0ne** [comment]
I've pushed an update that I think will fix the issue.
Mainly, I removed the built-in C# DirectoryServices method of finding the domain controller which is initializing the COM interface in the process before the tool has a chance to initialize it itself with its hooks and settings that it needs.

If my suspicions are correct, once run the updated version you will get an error stating that the tool could not find the domain controller FQDN and it will ask you to specify it manually with the **--DomainController** flag. I hope that once you do that the tool will run without the issue.

Let me know if that solves it :)
